gradle-plugin/src/main/java/com/squareup/anvil/plugin/IncrementalSignal.kt

```diff
@@ -1,9 +1,42 @@
 package com.squareup.anvil.plugin
 
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
 import org.gradle.api.services.BuildService
 import org.gradle.api.services.BuildServiceParameters.None
+import kotlin.random.Random
+
+/*
+ * The key is important. Gradle uses different class loaders for different modules when the plugin
+ * is applied through the new plugins { } syntax. If we'd use the same key, then the
+ * IncrementalSignal class would be different for the same entry and Gradle would throw an error.
+ *
+ * By using a random key per class loader we avoid this issue. Since this service isn't shared
+ * across modules and only within a module it's not causing any problems tasks only use the service
+ * for their project.
+ *
+ * We could also create one service per module, but that seems wasteful. When the Gradle plugin is
+ * added to classpath of the root project, then the key is always the same and all modules share
+ * a single build service and a single cache.
+ */
+private val KEY = "incrementalSignal-" + Random.nextLong()
```

Comment:
`UUID.randomUUID().toString` instead? It's guaranteed to be universally unique (though in a practical sense, Random is _technically_ just as likely to be unique).

Reply:
Yeah, I have no concerns here, 2^64 possible values should be more than enough. For some reason it felt wrong to import a Java class in this Kotlin project. That's the only reason why I changed it 😅